apps/studio/src/features/editing-experience/components/LinkEditor/LinkHrefEditor.tsx

```diff
@@ -1,84 +1,95 @@
 import type { ReactNode } from "react"
-import { useState } from "react"
 import {
   Box,
   FormControl,
-  HStack,
-  Icon,
-  Text,
-  useRadioGroup,
+  Input,
+  InputGroup,
+  InputLeftAddon,
 } from "@chakra-ui/react"
 import { FormLabel } from "@opengovsg/design-system-react"
 
-import type { LinkTypeMapping, LinkTypes } from "./constants"
-import { LinkTypeRadioCard } from "./LinkTypeRadioCard"
-import { LinkTypeRadioContent } from "./LinkTypeRadioContent"
-import { getLinkHrefType } from "./utils"
+import { useLinkEditor } from "./LinkEditorContext"
+import { LinkEditorRadio } from "./LinkEditorRadio"
+
+const HTTPS_PREFIX = "https://"
+type HttpsLink = `https://${string}`
+
+const generateHttpsLink = (data: string): HttpsLink => {
+  if (data.startsWith(HTTPS_PREFIX)) {
+    return data as HttpsLink
+  }
+
+  return `https://${data}`
+}
```

Comment:
nit - should we use `HTTPS_PREFIX` for the `https://${data}` too?

Reply:
fixed!